### PR TITLE
Boiler glow green/yellow based on ammo

### PR DESCRIPTION
**Pull request by BobSlim**
## About The Pull Request

Boilers glow a mix of green/yellow, based on the ammo they have of neurotoxin/corrosivetoxin. Also a sneaky bugfix, that extinguished boilers stop glowing.

## Why It's Good For The Game

Looks cool. Boilers glowing white seems strange.

Gives nearby xenos/marines some information on the composition of the boiler's ammo reserves.

## Changelog
:cl:
tweak: Boilers glow a mix of yellow/green, based on amount of neuro/corrosive ammo stored.
fix: Boilers don't stop glowing when they are extinguished
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm [hunk 1]
```diff
@@ -20,14 +20,18 @@
...
+
 // ***************************************
 // *********** Init
 // ***************************************
 /mob/living/carbon/xenomorph/boiler/Initialize()
 	. = ..()
-	set_light(BOILER_LUMINOSITY)
 	smoke = new /datum/effect_system/smoke_spread/xeno/acid(src)
 	ammo = GLOB.ammo_list[/datum/ammo/xeno/boiler_gas]
+	updateBoilerGlow()
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm:34]: Don't you want to set the luminosity to 0 here?

Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm:34]: oh wait this is init

**Comment by BobSlim** [comment]
not sure if boilers will ever have a base glow in the future. I coded it anyway. Not sure how the excess code will slow down the system. Will revert if not required.

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm [hunk 2]
```diff
@@ -20,14 +20,25 @@
 	var/corrosive_ammo = 0
 	var/neuro_ammo = 0
 
+/mob/living/carbon/xenomorph/boiler/proc/updateBoilerGlow()
+	var/current_ammo = corrosive_ammo + neuro_ammo
+	var/ammo_glow = BOILER_LUMINOSITY_AMMO * current_ammo
+	var/glow = BOILER_LUMINOSITY_BASE + ammo_glow
+	var/color = BOILER_LUMINOSITY_BASE_COLOR
+	if(current_ammo > 0)
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm:28]:
```suggestion
	if(current_ammo)
```

**Comment by spookydonut** [comment]
smh that is not correct documentation formatting

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm [hunk 3]
```diff
@@ -68,7 +68,8 @@
 /datum/action/xeno_action/create_boiler_bomb/action_activate()
 	var/mob/living/carbon/xenomorph/boiler/X = owner
 
-	if(X.selected_ability) //bombarding or something else.
+//bombarding or something else.
+	if(X.selected_ability)
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm:72]:
```suggestion
	if(X.selected_ability) //bombarding or something else.
```

File: code/__DEFINES/mobs.dm [hunk 4]
```diff
@@ -516,7 +516,12 @@ GLOBAL_LIST_INIT(xenoupgradetiers, list(XENO_UPGRADE_BASETYPE, XENO_UPGRADE_INVA
 #define DEFILER_STING_AMOUNT_RECURRING		10
 
 //Boiler defines
-#define BOILER_LUMINOSITY					0
+#define BOILER_LUMINOSITY_BASE						0
+#define BOILER_LUMINOSITY_BASE_COLOR				LIGHT_COLOR_GREEN
+//don't set this to 0. How much each 'piece' of ammo in reserve glows by.
+#define BOILER_LUMINOSITY_AMMO						1
```
Comment (spookydonut) [inline code/__DEFINES/mobs.dm:522]:
```suggestion
#define BOILER_LUMINOSITY_AMMO						1 //don't set this to 0. How much each 'piece' of ammo in reserve glows by.
```

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm [hunk 5]
```diff
@@ -13,21 +13,34 @@
 	tier = XENO_TIER_THREE
 	upgrade = XENO_UPGRADE_ZERO
 	gib_chance = 100
-	drag_delay = 6 //pulling a big dead xeno is hard
+	//pulling a big dead xeno is hard
+	drag_delay = 6
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm:17]:
```suggestion
	drag_delay = 6 //pulling a big dead xeno is hard
```

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/castedatum_boiler.dm [hunk 6]
```diff
@@ -8,7 +8,8 @@
 
 	tier = XENO_TIER_THREE
 	upgrade = XENO_UPGRADE_BASETYPE
-	wound_type = "boiler" //used to match appropriate wound overlays
+	//used to match appropriate wound overlays
+	wound_type = "boiler"
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/castedatum_boiler.dm:12]:
```suggestion
	wound_type = "boiler" //used to match appropriate wound overlays
```

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/castedatum_boiler.dm [hunk 7]
```diff
@@ -45,7 +46,8 @@
 
 	// *** Boiler Abilities *** //
 	max_ammo = 4
-	bomb_strength = 1 //Multiplier to the effectiveness of the boiler glob.
+	//Multiplier to the effectiveness of the boiler glob.
+	bomb_strength = 1
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/castedatum_boiler.dm:50]:
```suggestion
	bomb_strength = 1 //Multiplier to the effectiveness of the boiler glob.
```

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm [hunk 8]
```diff
@@ -13,21 +13,34 @@
...
+	var/color = BOILER_LUMINOSITY_BASE_COLOR
+	if(current_ammo)
+		var/ammo_color = BlendRGB(BOILER_LUMINOSITY_AMMO_CORROSIVE_COLOR, BOILER_LUMINOSITY_AMMO_NEUROTOXIN_COLOR, neuro_ammo/current_ammo)
+		color = BlendRGB(color, ammo_color, ammo_glow/glow)
+	set_light(glow, l_color = color)
+
 // ***************************************
 // *********** Init
 // ***************************************
+
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm:38]:
```suggestion
```

**Comment by spookydonut** [comment]
this is a good feature pr

**Pull request merged**